File: ClientApp/src/app/_models/message.ts
export interface Message {
    messageId: number;
    senderId: number;
    senderUserName: string;
    senderPhotoUrl: string;
    recipientId: number;
    recipientUserName: string;
    recipientPhotoUrl: string;
    content: string;
    dateMessageRead?: Date;
    dateMessageSent: Date;
}

File: Domain/Entities/Message.cs
namespace Domain.Entities
{
    public class Message
    {
        public int MessageId { get; set; }
        public int SenderId { get; set; }
        public string SenderUserName { get; set; }
        public AppUser Sender { get; set; }
        public int ReceipientId { get; set; }
        public string ReceipientUserName { get; set; }
        public AppUser Receipient { get; set; }
        public string Content { get; set; }
        public DateTime DateMessageSent { get; set; } = DateTime.UtcNow;
        public DateTime? DateMessageRead { get; set; }
        public bool SenderDeleted { get; set; }
        public bool ReceipientDeleted { get; set; }
    }
}
